src/shared/components/query/QueryStore.ts

```diff
@@ -116,36 +116,48 @@ export class QueryStore
 		molecularProfileIds: ReadonlyArray<string>
 	};
 
-	constructor(_window:Window, urlWithInitialParams?:string)
+	constructor(urlWithInitialParams?:string)
 	{
-		
+		this.initialize(urlWithInitialParams);
+	}
+
+	initialize(urlWithInitialParams?:string){
+
 		labelMobxPromises(this);
-		if (urlWithInitialParams)
-			this.setParamsFromUrl(urlWithInitialParams);
-
-        this.setParamsFromLocalStorage();
-
-		// need to create initialNonMolecularProfileParams using caseIds obtained from page, because
-		//	at this point asyncCustomCaseSet may not have resolved, and we dont want to wait for it
-		//	to resolve to create the initial query record, because theres a (remote) chance the user
-		//	could have modified the custom case set before the promise initially resolved, thus
-		//	making it an imperfect record of the initial query. Using this.caseIds instead of
-		//	asyncCustomCaseSet is the only failsafe way to make a copy of the initial query state
-		let initialNonMolecularProfileParams = nonMolecularProfileParams(this, this.caseIds);
-
-		this.initialQueryParams = {
-			nonMolecularProfileParams: initialNonMolecularProfileParams,
-			pathname: queryUrl(this, initialNonMolecularProfileParams),
-			molecularProfileIds: this._selectedProfileIds || []
-		};
 
-		reaction(
-			()=>this.allSelectedStudyIds,
-			()=>{
-				this.studiesHaveChangedSinceInitialization = true;
-			}
-		);
+        if (urlWithInitialParams)
+            this.setParamsFromUrl(urlWithInitialParams);
+
+        let legacySubmission: any = localStorage.getItem("legacyStudySubmission");
+        localStorage.removeItem("legacyStudySubmission");
+        if (legacySubmission) {
+            try{
+                this.setParamsFromLocalStorage(JSON.parse(legacySubmission));
+            } catch (ex) {
+                throw("error parseing/setting legacyStudySubmission");
+            }
+        }
 
+        // need to create initialNonMolecularProfileParams using caseIds obtained from page, because

```

Comment:
I think we can take this, and initialQueryParams, out entirely, now that we're doing the new thing with creating a new QueryStore each click with source of truth in the URL/session

Reply:
I'll check